src/model/writer.js

```diff
@@ -1016,6 +1016,35 @@ export default class Writer {
 		}
 	}
 
+	/**
+	 * Disables default gravity behaviour that tries to get attributes from nodes surrounding the caret. When gravity is
+	 * marked as overridden then attributes from the node before the caret won't be taken into consideration while
+	 * updating selection attributes.
+	 *
+	 * For the following model fragment:
+	 *
+	 * 		<$text bold="true" linkHref="url">bar[]</$text><$text bold="true">biz</$text>
+	 *
+	 * Selection attribute keys before override will be equal `[ 'bold', 'linkHref' ]`
+	 * Selection attribute keys after override will be equal `[ 'bold' ]`
+	 *
+	 * As default gravity is automatically restored just after a direct selection change event but this behaviour
+	 * can be disabled by passing `true` flag as param.
+	 *
+	 * @param {Boolean} [customRestore=false] When `true` then gravity won't be restored until
+	 * {@link ~Writer#overrideSelectionGravity} will be called directly.
+	 */
+	overrideSelectionGravity( customRestore ) {
+		this.model.document.selection._overrideGravity( customRestore );
+	}
+
+	/**
+	 * Restore overridden gravity to default.
+	 */
+	restoreSelectionGravity() {
```

Comment:
What would happen if we had two features which independently override and restore gravity? They might start to conflict with each other. Instead of:

1. A: override
1. A: restore
1. B: override
1. B: restore

The sequence might look like this:


1. A: override
1. B: override
1. A: restore
1. B: restore

In which case, the B feature's decision to override the gravity might be immediately cancelled by feature A which decided that it's time to restore the gravity.

What I'd propose is that `overrideSelectionGravity()` returned an identifier of "overriding" and this identifier will need to be passed to `restoreSelectionGravity()`.

Normally, I'd propose that `overrideSelectionGravity()` returned a callback, but we must enforce calling it in the `change()` block, so that's why the identifier.

Reply:
Ofc, the default gravity will be restored only after all features restored gravity.

Actually, this could be done even simpler by just having a counter of how many times `override()` was called. This could be ok too.